R/epichains.R

```diff
@@ -136,13 +138,33 @@ epichains_summary <- function(chains_summary,
                               index_cases,
                               statistic,
                               offspring_dist,
-                              stat_max) {
-  # Check that inputs are well specified
-  checkmate::assert_vector(chains_summary)
-  checkmate::assert_integerish(index_cases, null.ok = TRUE)
-  checkmate::assert_character(statistic)
+                              stat_max = Inf) {
+  # chain_summary can sometimes contain infinite values, so check
+  # that it's all infinite OR the non-infinite values are integerish.
+  # If it doesn't contain infinite values, then they are all integerish.
+  if (any(is.infinite(chains_summary))) {
+    checkmate::assert(
+      all(is.infinite(chains_summary)),
+      checkmate::check_integerish(
+        chains_summary[!is.infinite(chains_summary)],
+        lower = 0,
+        any.missing = FALSE
+      )
+    )
+  } else {
+    checkmate::assert_integerish(
+      chains_summary,
+      len = index_cases,
+      lower = 0
+    )
+  }
+  checkmate::assert_count(index_cases, positive = TRUE)
+  checkmate::assert_choice(statistic, choices = c("size", "length"))
```

Comment:
Just a suggestion to provide choices vector as argument default, same as above, as you're doing in `aggregate.epichains()`.

Reply:
This has been fixed in 3a97e40cc6e486362cd59d067e5b63ab2d1871a1.